tardis/workflows/standard_simulation_solver.py

```diff
@@ -0,0 +1,318 @@
+                    Path(configuration.config_dirname) / configuration.atom_data
+                )
+
+        else:
+            raise ValueError("No atom_data option found in the configuration.")
+
+        logger.info(f"\n\tReading Atomic Data from {atom_data_fname}")
+
+        try:
+            atom_data = AtomData.from_hdf(atom_data_fname)
+        except TypeError:
+            logger.exception(
+                "TypeError might be from the use of an old-format of the atomic database, \n"
+                "please see https://github.com/tardis-sn/tardis-refdata/tree/master/atom_data"
+                " for the most recent version.",
+            )
+            raise
+
+        return atom_data
+
+    def estimate_t_inner(
+        self,
+        input_t_inner,
+        luminosity_requested,
+        emitted_luminosity,
+        t_inner_update_exponent=-0.5,
+    ):
+        luminosity_ratios = (
+            (emitted_luminosity / luminosity_requested).to(1).value
+        )
+
+        return input_t_inner * luminosity_ratios**t_inner_update_exponent
+
+    def get_convergence_estimates(self, emitted_luminosity):
+        (
+            estimated_t_radiative,
+            estimated_dilution_factor,
+        ) = (
+            self.transport_solver.transport_state.calculate_radiationfield_properties()
+        )
+
+        estimated_t_inner = self.estimate_t_inner(
+            self.simulation_state.t_inner,
+            self.luminosity_requested,
+            emitted_luminosity,
+            t_inner_update_exponent=self.convergence_strategy.t_inner_update_exponent,
+        )
+        return {
+            "t_radiative": estimated_t_radiative,
+            "dilution_factor": estimated_dilution_factor,
+            "t_inner": estimated_t_inner,
+        }
+
+    def check_convergence(
+        self,
+        estimated_values,
+    ):
+        convergence_statuses = []
+
+        for key, solver in self.convergence_solvers.items():
+            current_value = getattr(self.simulation_state, key)
+            estimated_value = estimated_values[key]
+            no_of_shells = (
+                self.simulation_state.no_of_shells if key != "t_inner" else 1
+            )
+            convergence_statuses.append(
+                solver.get_convergence_status(
+                    current_value, estimated_value, no_of_shells
+                )
+            )
+
+        if np.all(convergence_statuses):
+            hold_iterations = self.convergence_strategy.hold_iterations
+            self.consecutive_converges_count += 1
+            logger.info(
+                f"Iteration converged {self.consecutive_converges_count:d}/{(hold_iterations + 1):d} consecutive "
+                f"times."
+            )
+            return self.consecutive_converges_count == hold_iterations + 1
+
+        self.consecutive_converges_count = 0
+        return False
+
+    def solve_simulation_state(
+        self,
+        estimated_values,
+    ):
```

Comment:
Given that we are *in* the simulation solver class, perhaps this should just be in `solve()`